### PR TITLE
feat: filter snap search query

### LINKED ISSUE #863: Search with special characters gives bad query
Poor title (again, sorry). Do we need an error message like the 🐣 ❓  or just allow special characters?
You can search with a **+** and **&** on _snapcraft.io_ although the results are the same as when you search without.

image

### PR DESCRIPTION
**Pull request by d-loose**
UDENG-1484
Fix #863

### CODE REVIEW

File: lib/src/snapd/snap_search.dart [hunk 1]
```diff
@@ -8,6 +8,13 @@ class SnapSearchParameters {
   final String? query;
   final SnapCategoryEnum? category;
 
+  // Removes characters from the search query that would cause snapd to reply
+  // with 'bad query'.
```
Comment (tim-hm) [inline lib/src/snapd/snap_search.dart:12]: Any chance we have a reference to this in the snapd docs?

File: test/snap_search_test.dart [hunk 2]
```diff
@@ -0,0 +1,12 @@
+import 'package:app_center/snapd.dart';
+import 'package:flutter_test/flutter_test.dart';
+
+void main() {
+  test('cleaned query', () {
+    const searchParamters = SnapSearchParameters(
+      query: 'a+b=c&d|e>f<g!h(i)j{k}l[m]n^o"p~q*r?s:t\\u/v',
+    );
+    expect(searchParamters.cleanedQuery,
+        equals('a b c d e f g h i j k l m n o p q r s t u v'));
```
Comment (tim-hm) [inline test/snap_search_test.dart:10]: Neat!

**Review by tim-hm — APPROVED** [review]
LGTM. Just one comment inline for your consideration.

File: lib/src/snapd/snap_search.dart (hunk 1 above)
Comment (d-loose) [reply lib/src/snapd/snap_search.dart:12]: Haven't found anything in the docs on this, but added the source code reference

**Pull request merged**